acceptance/tests/resource/package/yum.rb

```diff
@@ -1,186 +1,186 @@
+                         # actual changing of resources could irreparably damage a
+                         # host running this, or require special permissions.
 
-require 'puppet/acceptance/rpm_util'
-extend Puppet::Acceptance::RpmUtils
+  require 'puppet/acceptance/rpm_util'
+  extend Puppet::Acceptance::RpmUtils
 
-epoch_rpm_options    = {:pkg => 'epoch', :version => '1.1', :epoch => '1'}
-no_epoch_rpm_options = {:pkg => 'guid', :version => '1.0'}
+  epoch_rpm_options    = {:pkg => 'epoch', :version => '1.1', :epoch => '1'}
+  no_epoch_rpm_options = {:pkg => 'guid', :version => '1.0'}
 
-teardown do
-  step "cleanup"
-  agents.each do |agent|
-    clean_rpm agent, epoch_rpm_options
-    clean_rpm agent, no_epoch_rpm_options
-  end
-end
-
-def verify_state(hosts, pkg, state, match)
-  hosts.each do |agent|
-    cmd = rpm_provider(agent)
-    # Note yum and dnf list packages as <name>.<arch>
-    on agent, "#{cmd} list installed" do
-      method(match).call(/^#{pkg}\./, stdout)
+  teardown do
+    step "cleanup"
+    agents.each do |agent|
+      clean_rpm agent, epoch_rpm_options
+      clean_rpm agent, no_epoch_rpm_options
     end
   end
-end
-
-def verify_present(hosts, pkg)
-  verify_state(hosts, pkg, '(?!purged|absent)[^\']+', :assert_match)
-end
-
-def verify_absent(hosts, pkg)
-  verify_state(hosts, pkg, '(?:purged|absent)', :assert_no_match)
-end
-
-step "Managing a package which does not include an epoch in its version" do
-  step 'Setup repo and package'
-  agents.each do |agent|
-    clean_rpm agent, no_epoch_rpm_options
-    setup_rpm agent, no_epoch_rpm_options
-    send_rpm agent, no_epoch_rpm_options
-  end
 
-  step 'Installing a known package succeeds' do
-    verify_absent agents, 'guid'
-    apply_manifest_on(agents, 'package {"guid": ensure => installed}') do |result|
-      assert_match('Package[guid]/ensure: created', "#{result.host}: #{result.stdout}")
+  def verify_state(hosts, pkg, state, match)
+    hosts.each do |agent|
+      cmd = rpm_provider(agent)
+      # Note yum and dnf list packages as <name>.<arch>
+      on agent, "#{cmd} list installed" do
+        method(match).call(/^#{pkg}\./, stdout)
+      end
     end
   end
 
-  step 'Removing a known package succeeds' do
-    verify_present agents, 'guid'
-    apply_manifest_on(agents, 'package {"guid": ensure => absent}') do |result|
-      assert_match('Package[guid]/ensure: removed', "#{result.host}: #{result.stdout}")
-    end
+  def verify_present(hosts, pkg)
+    verify_state(hosts, pkg, '(?!purged|absent)[^\']+', :assert_match)
   end
 
-  step 'Installing a specific version of a known package succeeds' do
-    verify_absent agents, 'guid'
-    apply_manifest_on(agents, 'package {"guid": ensure => "1.0"}') do |result|
-      assert_match('Package[guid]/ensure: created', "#{result.host}: #{result.stdout}")
-    end
+  def verify_absent(hosts, pkg)
+    verify_state(hosts, pkg, '(?:purged|absent)', :assert_no_match)
```

Comment:
looks like these various `verify_*` methods should be abstracted into an acceptance library.